--- a/CMakeLists.txt	2020-05-16 14:08:50.801669540 +0000
+++ b/CMakeLists.txt	2020-05-16 14:09:43.609057340 +0000
@@ -13,6 +13,8 @@
 cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
 
 project(SOCI)
+include(GNUInstallDirs)
+message (STATUS "CMAKE_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR}")
 
 ###############################################################################
 # Build features and variants
@@ -134,11 +136,7 @@
 ###############################################################################
 
 if(NOT DEFINED SOCI_LIBDIR)
-  if(APPLE OR CMAKE_SIZEOF_VOID_P EQUAL 4)
-    set(SOCI_LIBDIR "lib")
-  else()
-    set(SOCI_LIBDIR "lib64")
-  endif()
+  set(SOCI_LIBDIR "${CMAKE_INSTALL_LIBDIR}")
 endif()
 
 set(BINDIR "bin" CACHE PATH "The directory to install binaries into.")
